torchmetrics/text/chrf.py

```diff
@@ -0,0 +1,210 @@
+    both ChrF score computation introduced in [1] and chrF++ score introduced in `chrF++ score_`. This
+    implementation follows the implmenetaions from https://github.com/m-popovic/chrF and
+    https://github.com/mjpost/sacrebleu/blob/master/sacrebleu/metrics/chrf.py.
+
+    Args:
+        n_char_order:
+            A character n-gram order. If `n_char_order=6`, the metrics refers to the official chrF/chrF++.
+        n_word_order:
+            A word n-gram order. If `n_word_order=2`, the metric refers to the official chrF++. If `n_word_order=0`, the
+            metric is equivalent to the original ChrF.
+        beta:
+            A parameter determining an importance of recall w.r.t. precision. If `beta=1`, their importance is equal.
+        lowercase:
+            An indication whether to enable case-insesitivity.
+        whitespace:
+            An indication whether keep whitespaces during n-gram extraction.
+        return_sentence_level_score:
+            An indication whether a sentence-level chrF/chrF++ score to be returned.
+        compute_on_step:
+            Forward only calls ``update()`` and returns None if this is set to False. default: True
+        dist_sync_on_step:
+            Synchronize metric state across processes at each ``forward()``
+            before returning the value at the step.
+        process_group:
+            Specify the process group on which synchronization is called. default: None (which selects the entire world)
+        dist_sync_fn:
+            Callback that performs the allgather operation on the metric state. When `None`, DDP
+            will be used to perform the allgather.
+
+    Example:
+        >>> hypothesis_corpus = ['the cat is on the mat']
+        >>> reference_corpus = [['there is a cat on the mat', 'a cat is on the mat']]
+        >>> metric = CHRFScore()
+        >>> metric(reference_corpus, hypothesis_corpus)
+        tensor(0.8640)
+
+    References:
+        [1] chrF: character n-gram F-score for automatic MT evaluation by Maja Popović `chrF score`_
+        [2] chrF++: words helping character n-grams by Maja Popović `chrF++ score`_
+    """
+
+    is_differentiable = False
+    higher_is_better = True
+    sentence_chrf_score: Optional[List[Tensor]] = None
+
+    def __init__(
+        self,
+        n_char_order: int = 6,
+        n_word_order: int = 2,
+        beta: float = 2.0,
+        lowercase: bool = False,
+        whitespace: bool = False,
+        return_sentence_level_score: bool = False,
+        compute_on_step: bool = True,
+        dist_sync_on_step: bool = False,
+        process_group: Optional[Any] = None,
+        dist_sync_fn: Optional[Callable] = None,
+    ):
+        super().__init__(
+            compute_on_step=compute_on_step,
+            dist_sync_on_step=dist_sync_on_step,
+            process_group=process_group,
+            dist_sync_fn=dist_sync_fn,
+        )
+
+        self.n_char_order = n_char_order
```

Comment:
lets add input validation :]
```suggestion
        if not isinstance(n_char_order, int) or n_char_order < 1:
            raise ValueError("Expected argument `n_char_order` to be an integer higher than 0")
        self.n_char_order = n_char_order
```